client/src/main/java/com/github/cameltooling/netbeans/client/preferences/ProcessPreferences.java

```diff
@@ -0,0 +1,109 @@
+    static final String CAMEL_CATALOG_VERSION_PREF_KEY = "Camel catalog version";
+    static final String CAMEL_ADDITIONAL_COMPONENT_PREF_KEY = "extra-components";
+    static final String CAMEL_CATALOG_RUNTIME_PROVIDER_PREF_KEY = "Camel catalog runtime provider";
+
+    /**
+     * Gets user preferences and returns them as JSON format accepted by Camel
+     * Langauge Server.
+     *
+     * @return User preferences in format accepted by Camel Language Server.
+     */
+    public static Map<String, Map<String, Object>> getPreferenceAsLanguageServerFormat() {
+        Map<String, Map<String, Object>> settings = new HashMap<>();
+        Map<String, Object> camelSettings = new HashMap<>();
+        camelSettings.put(CAMEL_CATALOG_VERSION_PREF_KEY, NbPreferences.forModule(ApacheCamelPanel.class).get("CatalogVersionPreference", Utilities.CATALOG_VERSION_DEFAULT)); // catalog version
+        camelSettings.put(CAMEL_CATALOG_RUNTIME_PROVIDER_PREF_KEY, getProviderAsString()); // runtime provider 
+        camelSettings.put(CAMEL_ADDITIONAL_COMPONENT_PREF_KEY, getAdditionalComponentIfValid()); // additional components
+        settings.put(TOP_NODE_CAMEL_KEY, camelSettings);
+        return settings;
+    }
+
+    /**
+     * Converts Catalog provider from preferences to String.
+     *
+     * @return Catalog provider as String.
+     */
+    private static String getProviderAsString() {
+        return switch (NbPreferences.forModule(ApacheCamelPanel.class).getInt("RuntimeProviderPreference", Utilities.CATALOG_PROVIDER_DEFAULT)) {
+            case 1 ->
+                "SPRINGBOOT";
+            case 2 ->
+                "QUARKUS";
+            case 3 ->
+                "KARAF";
+            default ->
+                "DEFAULT";
+        };
+    }
+
+    /**
+     * Gets Additional Component from user preferences and checks if they're
+     * valid. If so, they're added to list.
+     *
+     * @return List of components.
+     */
+    private static List<?> getAdditionalComponentIfValid() {
+        String additionalComponentAsString = NbPreferences.forModule(ApacheCamelPanel.class).get("AdditionalComponentPreference", Utilities.ADDITIONAL_COMPONENTS_DEFAULT);
+        if (additionalComponentAsString != null && !additionalComponentAsString.isEmpty()) {
+            try {
+                return new Gson().fromJson(additionalComponentAsString, List.class);
+            } catch (JsonSyntaxException ex) {
+                return Collections.emptyList();
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     *
+     */
+    public void sendDidChangeConfiguration() {
+        Map<String, Map<String, Object>> configuration = ProcessPreferences.getPreferenceAsLanguageServerFormat();
+        DidChangeConfigurationParams params = new DidChangeConfigurationParams(configuration); // create configuration
+        Set<LSPBindings> listBindings = LSPBindings.getAllBindings(); // get all runings LSP bindings
```

Comment:
typo runings


but I gfeel like the naming of the method is enoughto understand